### PR TITLE
31 add new task bottom sheet android

### PR DESCRIPTION
**Pull request by Njumbi**
Kindly review and merge

### CODE REVIEW

File: app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+                    }
+                }
+                true
+            }
+            popup.show()
+        }
+
+        binding.tvEndTask.setOnClickListener { dismiss() }
+    }
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt:83]:
```
binding.buttonAddTask.setOnClickListener {
            if (binding.editTextTaskName.text.isNullOrEmpty()) {
                Toast.makeText(requireContext(), "Task name cannot be empty", Toast.LENGTH_SHORT).show()
            } else {
                //get the task name, date and time
                val taskName = binding.editTextTaskName.text.toString()
                val taskDate = dateTime
                //create the task
                TodoTask(taskName, taskDate, taskStatus)
                //add the task to the database
                //dismiss the dialog
                dismiss()
            }
        }
```

File: app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt:84]:
```
private fun TodoTask(taskName: String, taskDate: String, taskStatus: String): Any {
//call API and pass the data
        return Toast.makeText(requireContext(), "Task name: $taskName, Date: $taskDate, Status: $taskStatus", Toast.LENGTH_SHORT).show()
    }
```

File: app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+open class BottomSheetFragment : BottomSheetDialogFragment() {
+    private lateinit var binding: FragmentBottomsheetBinding
+    // default date is today
+    private var dateTime = "Today"
+        set(value) {
+            binding.tvDatePicker.text = value
+            field = value
+        }
+
```
Comment (malcolmmaima) [inline app/src/main/java/com/todoist_android/ui/home/BottomSheetFragment.kt:32]:
```
//default taskStatus is created
    private var taskStatus = "Created"
```

**Review by malcolmmaima — CHANGES_REQUESTED** [review]
Left some comments/suggestions ... But this looks really good 👏

**Comment by Njumbi** [comment]
[Add New Task (Bottom Sheet) - Android](https://trello.com/c/Phnyjy4v/31-add-new-task-bottom-sheet-android)

**Comment by malcolmmaima** [comment]
Rebased to main

**Review by malcolmmaima — APPROVED** [review]

**Pull request merged**